contrib/kube-prometheus/examples/ingress.jsonnet

```diff
@@ -11,10 +11,71 @@ local kp =
     _config+:: {
       namespace: 'monitoring',
     },
+    // Configure External URL's per application
+    alertmanager+:: {
+      alertmanager+: {
+        spec+: {
+          externalURL: 'http://alertmanager.example.com',
+        },
+      },
+    },
+    grafana+:: {
+      config: {
```

Comment:
Please use the pluses all the way down so it merges with other configs.
```diff
-config: {
+config+: {
```

Reply:
This tripped me up initially when I started customizing Grafana. 

`config: null` is set [upstream](https://github.com/brancz/kubernetes-grafana/blob/master/grafana/grafana.libsonnet#L26) so if you don't override with `config: { ...` then building fails with the error:
`RUNTIME ERROR: binary operator + requires matching types, got null and object.`

I modified my local grafana.libsonnet to `config: {},` and this then allows use of pluses in `ingress.jsonnet` as types will match.
I think the Grafana config section [here](https://github.com/brancz/kubernetes-grafana/blob/master/grafana/grafana.libsonnet#L33) can be rewritten without the `!=null` (and match dashboards, etc). 

I will push commit here with pluses and open a PR to Grafana that is linked here.

Reply:
Unfortunately setting with pluses with a simplified `example.jsonnet` `_config` of: 
```
 _config+:: {                                                                                                                                                                                                                                                                                                               
    namespace: 'monitoring',                                                                                                                                                                                                                                                                                                 
    grafana+:: {                                                                                                                                                                                                                                                                                                             
      config+: {                                                                                                                                                                                                                                                                                                             
        sections+: {                                                                                                                                                                                                                                                                                                         
          server+: {                                                                                                                                                                                                                                                                                                         
            root_url: 'http://grafana.example.com/',                                                                                                                                                                                                                                                                         
          },                                                                                                                                                                                                                                                                                                                 
        },                                                                                                                                                                                                                                                                                                                   
      },                                                                                                                                                                                                                                                                                                                     
    },                                                                                                                                                                                                                                                                                                                       
  },                
```

results in the following error:
```
RUNTIME ERROR: binary operator + requires matching types, got null and object.
        example.jsonnet:(4:16)-(12:6)   object <anonymous>
        /mnt/data/github/my-kube-prometheus/vendor/grafana/grafana.libsonnet:33:9-33    thunk <a>
```

`config: null` set [here](https://github.com/brancz/kubernetes-grafana/blob/master/grafana/grafana.libsonnet#L26)

With the following changes to [`grafana.libsonnet`](https://github.com/brancz/kubernetes-grafana/blob/master/grafana/grafana.libsonnet) however we can compile with both empty and custom `Grafana.config`:

L26: `      config: {},`
L33: `[if std.length($._config.grafana.config) > 0 then 'config']:`

As this affects custom Grafana config for anyone, I think a PR to brancz's Grafana would be good.
The LDAP if statement could be changed in the same way. 

I've done a commit to this repo with the pluses and can do required PR to brancz's Grafana in the next few days.